cosmos/plugin/af3_plugin_impl.py

```diff
@@ -0,0 +1,419 @@
+from __future__ import annotations
+
+import json
+import mimetypes
+import os.path as op
+from typing import Any, Optional, Tuple
+
+from airflow.configuration import conf
+from airflow.plugins_manager import AirflowPlugin
+from fastapi import FastAPI
+from fastapi.responses import HTMLResponse, JSONResponse, Response
+from fastapi.staticfiles import StaticFiles
+
+
+def bucket_and_key(path: str) -> Tuple[str, str]:
+    from urllib.parse import urlsplit
+
+    parsed_url = urlsplit(path)
+    bucket = parsed_url.netloc
+    key = parsed_url.path.lstrip("/")
+    return bucket, key
+
+
+def open_s3_file(path: str, conn_id: Optional[str]) -> str:
```

Comment:
While talking to @dwreeves he suggested we may want to use https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/objectstorage.html instead of having specific functions per object storage